tests/Auth/UserManagerClusterTest.cpp

```diff
@@ -76,168 +89,83 @@ class UserManagerClusterTest : public ::testing::Test {
   EXPECT_EQ(um->globalVersion(), getAgencyUserVersion());
-
-  uint64_t versionBefore = getAgencyUserVersion();
-
-  um->triggerLocalReload();
-  EXPECT_EQ(versionBefore, getAgencyUserVersion());
-
-  /*
-   * This is the correct test here, see above
-   *
-   * this->simulateOneHeartbeat();
-   */
-  // This needs to trigger a reload from DB
-  try {
-    um->userExists("unknown user");
-    FAIL();
-  } catch (arangodb::basics::Exception const& e) {
-    // This Execption indicates that we got past the version
-    // checks and would contact DBServer now.
-    // This is not under test here, we only want to test that we load
-    // the plan
-    EXPECT_EQ(e.code(), TRI_ERROR_DEBUG);
-  } catch (...) {
-    FAIL();
-  }
 }
 
 TEST_F(UserManagerClusterTest, triggerGlobalReloadShouldUpdateClusterVersion) {
-  TRI_AddFailurePointDebugging(FailureOnLoadDB);
-  auto guard = arangodb::scopeGuard(
-      []() noexcept { TRI_RemoveFailurePointDebugging(FailureOnLoadDB); });
-
   auto um = userManager();
   // If for some reason this EXPECT ever triggers, we can
   // inject either the AgencyValue into the UserManager
-  // or vice-versa. This is just an assertion that we
+  // or vice versa. This is just an assertion that we
   // expect everything to start at default (1).
-  EXPECT_EQ(um->globalVersion(), getAgencyUserVersion());
-
-  uint64_t versionBefore = getAgencyUserVersion();
+  uint64_t const versionBeforeGlobalReload = getAgencyUserVersion();
+  EXPECT_EQ(um->globalVersion(), versionBeforeGlobalReload);
 
   um->triggerGlobalReload();
-  // The version in the Agency needs to be increased
-  EXPECT_LT(versionBefore, getAgencyUserVersion());
 
-  /*
-   * This is the correct test here, see above
-   *
-   * this->simulateOneHeartbeat();
-   */
-  // This needs to trigger a reload from DB
-  try {
-    um->userExists("unknown user");
-    FAIL();
-  } catch (arangodb::basics::Exception const& e) {
-    // This Execption indicates that we got past the version
-    // checks and would contact DBServer now.
-    // This is not under test here, we only want to test that we load
-    // the plan
-    EXPECT_EQ(e.code(), TRI_ERROR_DEBUG);
-  } catch (...) {
-    FAIL();
+  auto const versionAfterGlobalReload = getAgencyUserVersion();
+
+  // The version in the Agency needs to be increased
+  EXPECT_LT(versionBeforeGlobalReload, versionAfterGlobalReload);
+
+  // before the heartbeat we internally still have the state of
+  // global & internal version being equal, because no-one yet gave the new
+  // agency version to the user-manager
+  EXPECT_EQ(um->globalVersion(), um->internalVersion());
+  EXPECT_LT(um->globalVersion(), versionAfterGlobalReload);
+  EXPECT_LT(um->internalVersion(), versionAfterGlobalReload);
+
+  // Simulate heart beat
+  // This internally will increase the globalVersion and trigger the
+  // UpdateThread to start and preload the user cache
+  auto const globalVersionBeforeHeartbeat = um->globalVersion();
```

Comment:
We don't need `globalVersionBeforeHeartbeat`, it should be the same as `versionBeforeGlobalReload`